### ISSUE TITLE
VideoElementCache immediately fills when not on localhost

### ISSUE THREAD

**Issue by alex2** [bbc/VideoContext#48]
When using our application on a non-localhost domain, we immediately see
`No available video element in the cache, creating a new one. This may break mobile, make your initial cache larger.`
in the logs, which is intermittently followed immediately by the video (canvas) playback stopping.

On localhost we never experience the message or the video playback problem. We're using hash-based routing which might be the culprit, but we're using that on localhost too.

There's a line of code that inspects the url of the window so I suspect that might be the problem ¯\_(ツ)_/¯
```
if ((element.src === "" || element.src === undefined || element.src === stripHash(window.location)) && element.srcObject == null )return element;
```

**Comment by MatthewShotton** [comment]
I've been having a go at re-creating this but can't seem to manage it. Would you be able to send me an example of the URL structure?

The video element cache exists to enable support for mobile devices. If that's not a priority it's possible to disable it which might help for the time being. This can be done in the constructor for the VideoContext instance like so:

```
const ctx = new VideoContext(canvas, ()=>{}, {useVideoElementCache: false});
```
You'll need to update to v0.50.2 of the VideoContext library as while writing this i found a bug in the code which disables the cache.

**Comment by MatthewShotton** [comment]
This should be fixed in v0.50.3. The issue was the stripHash function adding a colon in for a port even when ```window.location.port === ""```.
